File: Array_String/Best_Time_to_Buy_and_Sell_Stock.java
class Solution {
    public int maxProfit(int[] prices)
    {
        int min_price = prices[0];
        int max_profit = 0;

        for (int i = 1; i < prices.length; i++)
        {
            if (prices[i] < min_price)
                min_price = prices[i];
            else
            {
                if (max_profit < prices[i] - min_price)
                    max_profit = prices[i] - min_price;
            }
        }
        return (max_profit);
    }

    public static void main(String[] args)
    {
        Solution solution = new Solution();
        int rtn_profit;

        int[] nums = new int[]{7,1,5,3,6,4};

        rtn_profit = solution.maxProfit(nums);
        System.out.println("Profit earned: " + rtn_profit);
    }
};

File: Python_bootcamp/Greedy/Best_time_to_buy_and_sell_stock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float('inf')
        max_profit = 0

        for price in prices:
            if price < min_price:
                min_price = price
            elif price - min_price > max_profit:
                max_profit = price - min_price
        return max_profit

if __name__ == "__main__":
    solution = Solution()
    nums = [7,1,5,3,6,4]
    print(solution.maxProfit(nums))